### PR TITLE
[Bugfix] fix deletion of results

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

For some reason the endpoints in the client didn't match the endpoints in the server anymore, 
which made deletion of results impossible. This is fixed now.
image

### Testing steps
1. course overview -> exercises page
2. for each exercise type click on the submissions button. From this page, navigate to the page for individual submissions. (Click the number in the submissions column)
3. If there is result for the submission it should be visible now, and a 'delete' button should be visible. Using this button should work for all exercise types
image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java [hunk 1]
```diff
@@ -298,7 +298,7 @@ else if (!result.getParticipation().getId().equals(participationId)) {
      * @param resultId     - the id of the result which should get deleted
      * @return 200 Ok response if canceling was successful, 403 Forbidden if current user is not an instructor of the course or an admin
      */
-    @DeleteMapping("participations/{participationId}/submissions/{submissionId}/results/{resultId}")
+    @DeleteMapping("participations/{participationId}/text-submissions/{submissionId}/delete/{resultId}")
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java:301]:
```suggestion
    @DeleteMapping("participations/{participationId}/text-submissions/{submissionId}/results/{resultId}")
```

Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java:301]: I don't think we should use `delete` here as part of the URL.
This should also be changed in the other cases

File: src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.service.ts [hunk 2]
```diff
@@ -47,11 +47,12 @@ export class FileUploadAssessmentService {
...
     /**
      * Deletes an assessment.
+     * @param participationId id of the participation, to which the assessment and the submission belong to
      * @param submissionId id of the submission, to which the assessment belongs to
      * @param resultId     id of the result which is deleted
      */
-    deleteAssessment(submissionId: number, resultId: number): Observable<void> {
-        return this.http.delete<void>(`${this.resourceUrl}/file-upload-submissions/${submissionId}/delete/${resultId}`);
+    deleteAssessment(participationId: number, submissionId: number, resultId: number): Observable<void> {
+        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/file-upload-submissions/${submissionId}/delete/${resultId}`);
```
Comment (krusche) [inline src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.service.ts:55]: I don't see those changes reflected on the server

Comment (krusche) [reply src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.service.ts:55]:
```suggestion
        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/file-upload-submissions/${submissionId}/results/${resultId}`);
```

File: src/main/webapp/app/exercises/modeling/assess/modeling-assessment.service.ts [hunk 3]
```diff
@@ -74,12 +74,13 @@ export class ModelingAssessmentService {
...
-     * Deletes an assessment.
+     * Deletes an assessment
+     * @param participationId id of the participation, to which the assessment and the submission belong to
      * @param submissionId id of the submission, to which the assessment belongs to
      * @param resultId     id of the result which is deleted
      */
-    deleteAssessment(submissionId: number, resultId: number): Observable<void> {
-        return this.http.delete<void>(`${this.resourceUrl}/modeling-submissions/${submissionId}/delete/${resultId}`);
+    deleteAssessment(participationId: number, submissionId: number, resultId: number): Observable<void> {
+        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/modeling-submissions/${submissionId}/delete/${resultId}`);
```
Comment (krusche) [inline src/main/webapp/app/exercises/modeling/assess/modeling-assessment.service.ts:83]:
```suggestion
        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/modeling-submissions/${submissionId}/results/${resultId}`);
```

Comment (krusche) [reply src/main/webapp/app/exercises/modeling/assess/modeling-assessment.service.ts:83]: I don't see those changes reflected on the server

File: src/main/webapp/app/exercises/programming/assess/manual-result/programming-assessment-manual-result.service.ts [hunk 4]
```diff
@@ -55,10 +55,11 @@ export class ProgrammingAssessmentManualResultService {
...
     /**
      * Deletes an assessment.
+     * @param participationId id of the participation, to which the assessment and the submission belong to
      * @param submissionId id of the submission, to which the assessment belongs to
      * @param resultId     id of the result which is deleted
      */
-    deleteAssessment(submissionId: number, resultId: number): Observable<void> {
-        return this.http.delete<void>(`${this.resourceUrl}/programming-submissions/${submissionId}/delete/${resultId}`);
+    deleteAssessment(participationId: number, submissionId: number, resultId: number): Observable<void> {
+        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/programming-submissions/${submissionId}/delete/${resultId}`);
```
Comment (krusche) [inline src/main/webapp/app/exercises/programming/assess/manual-result/programming-assessment-manual-result.service.ts:63]:
```suggestion
        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/programming-submissions/${submissionId}/results/${resultId}`);
```

Comment (krusche) [reply src/main/webapp/app/exercises/programming/assess/manual-result/programming-assessment-manual-result.service.ts:63]: I don't see those changes reflected on the server and in the server integration tests (they fail at the moment)

File: src/main/webapp/app/exercises/text/assess/text-assessment.service.ts [hunk 5]
```diff
@@ -95,15 +95,17 @@ export class TextAssessmentService {
...
     /**
      * Deletes an assessment.
+     * @param participationId id of the participation, to which the assessment and the submission belong to
      * @param submissionId id of the submission, to which the assessment belongs to
      * @param resultId     id of the result which is deleted
      */
-    deleteAssessment(submissionId: number, resultId: number): Observable<void> {
-        return this.http.delete<void>(`${this.resourceUrl}/text-submissions/${submissionId}/delete/${resultId}`);
+    deleteAssessment(participationId: number, submissionId: number, resultId: number): Observable<void> {
+        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/text-submissions/${submissionId}/delete/${resultId}`);
```
Comment (krusche) [inline src/main/webapp/app/exercises/text/assess/text-assessment.service.ts:103]:
```suggestion
        return this.http.delete<void>(`${this.resourceUrl}/participations/${participationId}/text-submissions/${submissionId}/results/${resultId}`);
```

Comment (krusche) [reply src/main/webapp/app/exercises/text/assess/text-assessment.service.ts:103]: I don't see those changes reflected on the server and in the server integration tests (they fail at the moment)

File: src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java (hunk 1 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java:301]: I don't see those changes reflected in the server integration tests (they fail at the moment)

**Review by krusche — CHANGES_REQUESTED** [review]
see my inline comments

**Review by FuchsDominik — DISMISSED** [review]
worked while testing 👍

**Review by jhainzlschmid — APPROVED** [review]
Tested in testing session and worked

**Review by canyelten — APPROVED** [review]
Tested on TS1, the delete button is visible and works as expected.

**Review by krusche — APPROVED** [review]
code looks good now 👍

**Pull request merged**